alacritty_terminal/src/term/mod.rs

```diff
@@ -1317,20 +1316,20 @@ impl<T: EventListener> Handler for Term<T> {
         }
 
         if width == 1 {
-            self.write_cursor_cell(c);
+            self.write_at_cursor(c);
         } else if width == 2 {
-            // Insert placeholder and linebreak if only one cell left for wide glyph
+            // Insert extra placeholder before wide char if glyph doesn't fit this row anymore
```

Comment:
nit: `glyph doesn't fit this row anymore` -> `glyph doesn't fit *in* this row anymore`